src/mobile/src/components/PostCard.jsx

```diff
@@ -0,0 +1,86 @@
+import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
+import { Entypo } from '@expo/vector-icons';
+import useSWR from 'swr';
+
+import Skeleton from './Skeleton';
+import { formatPublishedDate } from '../utils/formatPublishedDate';
+
+const styles = StyleSheet.create({
+  card: {
+    marginLeft: 16,
+    marginRight: 16,
+    padding: 12,
+  },
+  cardContent: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  cardText: {
+    flexShrink: 1,
+    marginLeft: 12,
+  },
+  cardTextError: {
+    color: '#EF4444',
+  },
+  cardPostInfo: {
+    marginTop: 8,
+    color: '#9CA3AF',
+    fontSize: 12,
+  },
+});
+
+const PostCard = ({ postURL, page, navigation }) => {
+  const { data: post, error } = useSWR(postURL);
```

Comment:
Please check this part, my network log shows it's running into an infinite loop.

Reply:
Not sure why but it seems that the way we set up navigation affect this so we end up having infinite loop

Reply:
On Mobile I don't see that issue tho, only using the web version having this bug.

Reply:
This bug can be fixed in https://github.com/Seneca-CDOT/telescope/pull/3599

Reply:
> The Flatlist  is not scrollable, making onEndReached is constantly called and makes size increment which forces useSWRInfinite to continue .
>If you add flex: 1 to its parent SafeAreaView , then it will work
> _Duc Bui Manh_